### PR TITLE
Tooling update

### PR DESCRIPTION
**Pull request by pozil**
- Split CI workflows for cleaner badge (added a dedicated PR workflow to avoid CI badge pollution)
- Bumped dependencies
- Updated linting rules (removed old rules overrides and disabled a new Salesforce-specific rule)

### CODE REVIEW

**Comment by pozil** [comment]
@muenzpraeger looks like the Heroku deploy is failing because Heroku is using version of Node that's too recent for us
```
Resolving node version >=10.0.0...
Downloading and installing node 14.4.0...
```

**Comment by muenzpraeger** [comment]
I’ll take a look tomorrow @pozil as if I can fix that on the `lwc-services` side of things. If not we specify them in `package.json`.

@hunterloftis the [Heroku docs](https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version) state node12 as current version for Heroku. Has that changed?

**Comment by muenzpraeger** [comment]
The module directory for local directories wasn't correctly configured. I move then `lwc` config out of `package.json` into `lwc.config.json` for better visibility.

File: .github/workflows/ci-pr.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                  path: node_modules
+                  key: yarn-${{ hashFiles('**/yarn.lock') }}
+                  restore-keys: |
+                      yarn-${{ env.cache-name }}-
+                      yarn-
+
+            # Install npm dependencies for Prettier and Jest
+            - name: 'Install npm dependencies'
+              if: steps.cache-yarn.outputs.cache-hit != 'true'
+              run: yarn install
```
Comment (muenzpraeger) [inline .github/workflows/ci-pr.yml:32]: We should change this to `yarn install --frozen-lockfile`, the counterpart of `npm ci`.

File: .github/workflows/ci-pr.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Unique name for this workflow
+name: CI (PR)
+
+# Definition when the workflow should run
+on:
+    pull_request:
+        types: [opened, synchronize, reopened]
```
Comment (muenzpraeger) [inline .github/workflows/ci-pr.yml:7]: Please add `edited`.

Comment (pozil) [reply .github/workflows/ci-pr.yml:7]: Good point, we need to fix that for all samples apps.
The orignal CI flow was missing it and I noticed that some workflows from other apps are missing that part too.

**Pull request merged**